### ISSUE TITLE
DataTable - Export to Excel

### ISSUE THREAD

**Issue by lousel** [wet-boew/wet-boew#7599]
I have been trying to get the DataTables built-in buttons working in the WET4 template, specifically the export to excel and haven't had any luck. I tried everything but can't get it to display the button.
I created a JSFiddle from the example off the DataTables website, if I use this code only the table is displayed no buttons: [https://jsfiddle.net/sk6e9311/1/](url)

So I am curious if I am doing something wrong, or if maybe the buttons are not supported?

Thank you!

**Comment by silentcanadian** [comment]
Here's a JSFiddle I've created which works.  I had to assign file names this way because they weren't being captured otherwise: https://jsfiddle.net/szb0fr1v/

**Comment by hsrudnicki** [comment]
Can be this closed as resolved or otherwise? Please update the status of this, whether this is still an issue?
Thx

**Comment by masterbee** [comment]
@lousel - we have a few issues with Datatables that we are looking into. In short the buttons are a bit of a fringe functionality that were not well adopted in the beginning due to some concerns. So WIP at the moment and we will glad take a look at any P.R.'s that you can think up to help integrate buttons flexibly.

**Comment by kodecount** [comment]
@masterbee Not sure really a bug. The data-table button functions well work, but we need to add the `js` for them to the [load function](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/tables/tables.js#L81) in the `tables.js`. I found that current way of loading the `js` files can slow the data-table speed though.

**Comment by kodecount** [comment]
@masterbee Not sure really a bug. The data-table button functions work, but we need to add the datatable button script to the [load function](https://github.com/wet-boew/wet-boew/blob/master/src/plugins/tables/tables.js#L81) in the `tables.js`. This goes the same for data-table printing, but I found this to slow data-table speeds.

SIDE NOTE: Also styling would need to be done on the generate buttons.

**Comment by LaurentGoderre** [comment]
This snippet shows how you can extend datatable without changing the WET code.

https://github.com/open-data/ckanext-canada/blob/master/ckanext/canada/templates/internal/recombinant/resource_edit.html#L37-L57

**Comment by kodecount** [comment]
@LaurentGoderre Yes. But still need to load the extra JavaScript script file which was causing the table to flicker for me.

**Comment by kodecount** [comment]
@lousel @masterbee Using @LaurentGoderre example works:

```javascript
    wb.doc.on("wb-ready.wb-tables", function() {
      Modernizr.load( {
        load: [
          "https://cdn.datatables.net/buttons/1.2.2/js/dataTables.buttons" + wb.getMode() + ".js",
          "https://cdn.datatables.net/buttons/1.2.2/js/buttons.html5" + wb.getMode() + ".js",
          "https://cdn.datatables.net/buttons/1.2.2/js/buttons.flash" + wb.getMode() + ".js",
          "https://cdnjs.cloudflare.com/ajax/libs/jszip/2.5.0/jszip" + wb.getMode() + ".js",
        ],
        complete: function() {
          var selector = "#table", table;
          if ($.fn.DataTable.isDataTable( selector ) ) {
            table = $( selector ).DataTable();
            new $.fn.dataTable.Buttons( table, {
              buttons: [ 'excel' ]
            } );
            table.buttons().container().appendTo( $( table.table().container() ).find( ".top" ) );
          }
        }
      });
     });
```
